src/dmd/dtemplate.d

```diff
@@ -537,6 +539,9 @@ extern (C++) final class TemplateDeclaration : ScopeDsymbol
     // threaded list of previous instantiation attempts on stack
     TemplatePrevious* previous;
 
+    private Expression lastConstraint; // the last failed constraint instantiation
+    private Array!Expression lastConstraintNegs; // its negative parts
```

Comment:
These should probably go in `TemplateInstance`, not declaration.

Reply:
Probably, but I don't see how we can retrieve those instances for overloads in `printCandidates`.